### PR TITLE
Support asm cloud services (Targeting latest branch)

### PR DESCRIPTION
**Pull request by nshampur**
Several large enterprises still extensively use Azure PaaS V1 (Cloud Services a.k.a. Web/Worker Roles) through the classic management API (ASM). These are early adopters of the Azure cloud platform and will take singificant migration effort to move to newer Compute options available via Azure Resource Manager (ARM).
Newer cross platform automation options (azure cli, Az Powershell Module) do not provide support for ASM, which makes it non feasible to run on non windows environment (e.g:- Linux Containers as part of CI/CD).
This pull request wraps the ASM API and adds support for the common workflows such as deploying cloud service packages to production or staging slots of a cloud service and performing blue/green style deployment with VIP Swap. It also adds support for managing extensions for cloud services which are needed for enabling extension such as PaaS Diagnostics (a.k.a. WAD a.k.a. Windows Azure Diagnostics) and Microsoft AntiMalware extensions.

Targeting latest branch

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Sadly we don't have tests for these packages.  Were you able to locally test your additions?

File: services/classic/management/virtualmachine/client.go [hunk 1]
```diff
@@ -89,7 +311,33 @@ func (vm VirtualMachineClient) GetDeploymentName(cloudServiceName string) (strin
 	if cloudServiceName == "" {
 		return "", fmt.Errorf(errParamNotSpecified, "cloudServiceName")
 	}
-	requestURL := fmt.Sprintf(azureListDeploymentsInSlotURL, cloudServiceName)
+	requestURL := fmt.Sprintf(azureDeploymentSlotURL, cloudServiceName, "Production")
```
Comment (jhendrixMSFT) [inline services/classic/management/virtualmachine/client.go]: Shouldn't the third param be `DeploymentSlotProduction`?

Comment (jhendrixMSFT) [reply services/classic/management/virtualmachine/client.go]: Or perhaps `GetDeploymentName()` calls `GetDeploymentNameForSlot()`?

File: services/classic/management/virtualmachine/entities.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ type DeploymentResponse struct {
 	ExtendedProperties     []ExtendedProperty `xml:">ExtendedProperty"`
 	PersistentVMDowntime   PersistentVMDowntime
 	VirtualIPs             []VirtualIP `xml:">VirtualIP"`
-	ExtensionConfiguration string      // cloud service extensions not fully implemented
+	ExtensionConfiguration ExtensionConfiguration
```
Comment (jhendrixMSFT) [inline services/classic/management/virtualmachine/entities.go]: This is a breaking change so we'll have to release this as part of v25 at the end of this month.

**Comment by nshampur** [comment]
@jhendrixMSFT .. Yes.  Tested it fairly well and even integrated into our CI/CD pipeline

File: services/classic/management/virtualmachine/entities.go (hunk 2 above)
Comment (nshampur) [reply services/classic/management/virtualmachine/entities.go]: @jhendrixMSFT . Sounds good if we need to wait for v25, even though the property would always be null previously.

File: services/classic/management/virtualmachine/client.go (hunk 1 above)
Comment (nshampur) [reply services/classic/management/virtualmachine/client.go]: good suggestions. I'll make the changes

**Comment by jhendrixMSFT** [comment]
@nshampur thanks for your contribution.

**Pull request merged**